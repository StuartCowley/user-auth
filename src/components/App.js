import React, { useState } from "react";
import { BrowserRouter as Router, Routes, Route, NavLink } from "react-router-dom";
import Home from "./Home";
import LoginForm from "./LoginForm";
import "../styles/app.css";
import SignupForm from "./SignupForm";

const App = () => {
  const [user, setUser] = useState({});

  return (
    <div className="app">
      <Router>
        <nav className="app__navbar">
          <NavLink className="app__nav-item" to="/">Home</NavLink>
          <NavLink className="app__nav-item" to="signup">Signup</NavLink>
          <NavLink className="app__nav-item" to="login">Login</NavLink>
        </nav>
        <Routes>
          <Route path="/" element={<Home user={user} />} />
          <Route path="/login" element={<LoginForm />} />
          <Route path="/signup" element={<SignupForm setUser={setUser} />} />
        </Routes>
      </Router>
    </div>

  );
}

export default App;
