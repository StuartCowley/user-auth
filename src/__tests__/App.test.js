import renderer from "react-test-renderer";
import App from '../components/App';

describe("App component", () => {
  it("matches snapshot", () => {
    const generated = renderer
      .create(<App />)
      .toJSON();

    expect(generated).toMatchSnapshot();
  })
});