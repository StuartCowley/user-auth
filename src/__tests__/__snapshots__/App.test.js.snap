// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div
  className="app"
>
  <nav
    className="app__navbar"
  >
    <a
      aria-current="page"
      className="app__nav-item active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="app__nav-item"
      href="/signup"
      onClick={[Function]}
    >
      Signup
    </a>
    <a
      className="app__nav-item"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </nav>
  <div
    className="home"
  >
    <h2>
      You are not logged in, 
      <a
        href="/login"
        onClick={[Function]}
      >
        log in
      </a>
       or 
      <a
        href="/signup"
        onClick={[Function]}
      >
        signup
      </a>
    </h2>
  </div>
</div>
`;
